new-frontiers-14/frontier-station-14
# Expeditions: teardown resources on failures to generate.

## About the PR



Tears down a map on expedition generation failure.

## Why / Balance


Needless accumulation of resources if failures occur, causes server lag.

## How to test


1. Add code to throw an exception in SpawnSalvageMissionJob.cs between lines 250 and 275 (after dungeon generation but before FTL starts). `throw new Exception();` is fine.
2. Buy an expedition ship (any).
3. Spawn a debug "salvage expeditions computer".
4. Select a mission.  It should take a while to generate a dungeon, then cancel.
5. Your buttons should remain clickable when it returns.
6. Check the admin objects panel (F7->Objects), you should not have errant expedition maps remaining.

## Media


## Requirements


## Breaking changes


**Changelog**



:cl:
- fix: Failing expeditions should not cause server lag - if an expedition fails to generate, please try another one.
